Frame.cc

```diff
@@ -961,16 +961,12 @@ bool Frame::FillSpectralProfileData(
                         profile_ok = true;
                         return profile_ok;
                     }
-                    bool use_swizzled_data(false);
-                    try {
-                        // check is the region mask valid (outside the lattice or not)
-                        region->XyMask();
-                        // if region mask is valid, then check is swizzled data available
-                        std::unique_lock<std::mutex> guard(_image_mutex);
-                        use_swizzled_data = _loader->UseRegionSpectralData(region->XyMask());
-                        guard.unlock();
-                    } catch (casacore::AipsError& err) {
-                        std::cerr << err.getMesg() << std::endl;
+                    bool region_mask_ok;
+                    bool use_swizzled_data;
+                    std::unique_lock<std::mutex> guard(_image_mutex);
+                    use_swizzled_data = _loader->UseRegionSpectralData(region->XyMask(), region_mask_ok);
```

Comment:
this seems rather involved, adding an additional parameter that just gets set to false if the mask is a null pointer. Why not just keep that in this function, by using something like 
```
std::unique_lock<std::mutex> guard(_image_mutex);
const casacore::ArrayLattice<casacore::Bool>* mask = region->XyMask();
guard.unlock();
if (mask) {
    use_swizled_data = _loader->UseRegionSpectralData(mask);
}
if (!mask) { // if region mask not valid, send a NaN to the frontend    
    CARTA::SpectralProfileData profile_data;
    profile_data.set_stokes(curr_stokes);
    ...
}
```